src/components/ReceiveMyAddress.js

```diff
@@ -23,7 +23,8 @@ export default function ReceiveMyAddress() {
   const lastSharedAddress = useSelector((state) => state.lastSharedAddress);
 
   const getNextAddress = async () => {
-    const { address, index } = await wallet.getNextAddress();
+    // Fetch the address but do not mark it as used yet.
+    const { address, index } = await wallet.getCurrentAddress();
```

Comment:
I don't understand this change. This method is to get the next address, not the current address.

Reply:
As mentioned above, the method [`getNextAddress()`](https://github.com/HathorNetwork/hathor-wallet-lib/blob/bfc290175b8ea113f8e5e624690d3aa15e2b6da8/src/new/wallet.js#L563-L565) not only fetches the next available address, but also marks the current one as used.
```ts
  async getNextAddress() {
    // First we mark the current address as used, then return the next
    await this.getCurrentAddress({ markAsUsed: true });
```

 That was the reason the QA #318 brought this issue up on item 9. By changing the call to `getCurrentAddress()` we don't do this marking and but otherwise keep the current behavior intact.